### PR TITLE
HBW-244 Add hooks callback

### PR DESCRIPTION
**Pull request by GideonEXE**

### CODE REVIEW

**Comment by GideonEXE** [comment]
Screenshot 2020-01-16 at 17 54 36
Screenshot 2020-01-16 at 17 54 46

**Comment by GideonEXE** [comment]
Screenshot 2020-01-16 at 17 54 36
Screenshot 2020-01-16 at 17 54 46
Screenshot 2020-01-16 at 17 54 59

**Comment by GideonEXE** [comment]
Screenshot 2020-01-16 at 18 40 36
Screenshot 2020-01-16 at 18 40 44
Screenshot 2020-01-16 at 18 41 03

File: hbw/app/models/hbw/task.rb [hunk 1]
```diff
@@ -25,6 +25,11 @@ def with_user(email)
         end
       end
 
+      def get_task_by_id(id)
+        do_request(:get,
+                   "task/#{id}")
```
Comment (TimAle) [inline hbw/app/models/hbw/task.rb]: It's OK to write this on one line :).

File: hbw/app/services/hbw/task_notifier.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+module HBW
+  class TaskNotifier
+    class << self
+      def call(widget, task_id, event_name, users = [])
+        cache_key = "#{task_id}_#{DateTime.current.strftime('%Q')}_#{SecureRandom.hex(10)}"
```
Comment (TimAle) [inline hbw/app/services/hbw/task_notifier.rb]: Are you sure we need `DateTime.current` from `ActiveSupport`? I think we can use native `DateTime.now` here.

File: hbw/app/services/hbw/task_notifier.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+          message = {
+            task_id: task_id,
+            event_name: event_name,
+            cache_key: cache_key
+          }
+
+          ActionCable.server.broadcast("task_channel_#{user}", message.to_json)
+        end
+
+        cache_key
```
Comment (TimAle) [inline hbw/app/services/hbw/task_notifier.rb]: Why?

File: hbw/config/routes.rb [hunk 4]
```diff
@@ -17,6 +17,16 @@
       get :check, on: :collection, action: :check_user
     end
 
+    resources :tasks, only: [:index] do
+      put :form, on: :member, action: :submit
+      get :lookup, on: :member
+      post :claim, on: :member, action: :claim
+    end
```
Comment (TimAle) [inline hbw/config/routes.rb]: Looks like duplicated lines 3-7.

File: hbw/config/routes.rb [hunk 5]
```diff
@@ -17,6 +17,16 @@
...
+
     post 'file_upload', to: 'files#upload'
+
+    namespace :events, defaults: { format: :json } do
+      resources :tasks
```
Comment (TimAle) [inline hbw/config/routes.rb]: `only: [:update]`

File: hbw/config/routes.rb [hunk 6]
```diff
@@ -17,6 +17,16 @@
...
 
+    resources :tasks, only: [:index] do
+      put :form, on: :member, action: :submit
+      get :lookup, on: :member
+      post :claim, on: :member, action: :claim
+    end
+
     post 'file_upload', to: 'files#upload'
+
+    namespace :events, defaults: { format: :json } do
```
Comment (TimAle) [inline hbw/config/routes.rb]: Doesn't `defaults` provided from upper level?

**Pull request merged**